### ISSUE TITLE
[PROPOSAL] Exclude items while syncing

### ISSUE THREAD

**Issue by raffaem** [SamSchott/maestral#303]
**Is your feature request related to a problem? Please describe.**
When you are syncing a large library, you mean realize Maestral is syncing items you no longer need.

You cannot actually do that:

![image](https://user-images.githubusercontent.com/54762742/106906436-74f67500-66fd-11eb-954c-bfb872044b15.png)

**Describe the solution you'd like**
Allow to exclude items while you are syncing

**Describe alternatives you've considered**
I think the standard Dropbox client allows you to do that?

**Additional context**

**Comment by SamSchott** [comment]
That used to be possible before v1.3.0 but could result in issues when a user would exclude and immediately include a folder again while it was being downloaded. At the time, I decided for the brute-force fix of preventing users to change the excluded list while syncing. This is admittedly not the most elegant solution but I decided that it's acceptable since there is an option to exclude files or folders *before* the initial download.

**Comment by SamSchott** [comment]
Agreed, it would be good to revisit this. It will involve some changes to how the exclusion list is handled.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/samschott/maestral/issues/303#issue-801335716

This appears similar to https://github.com/samschott/maestral/issues/958#issue-1840988001, @raffaem.
